commands/audit/audit.go

```diff
@@ -281,9 +282,13 @@ func createJasScansTasks(auditParallelRunner *utils.SecurityParallelRunner, scan
 				SignedDescriptions:          auditParams.OutputFormat() == format.Sarif,
 				ScanResults:                 targetResult,
 				TargetOutputDir:             auditParams.scanResultsOutputDir,
+				AllowPartialResults:         auditParams.AllowPartialResults(),
 			}
-			if generalError := runner.AddJasScannersTasks(params); generalError != nil {
+			if generalError = runner.AddJasScannersTasks(params); generalError != nil {
+				// TODO this fix was in order to avoid capturing the error twice when using partial-results. if this is disables the error is collected twice - once from the target error and once from general error
 				_ = targetResult.AddTargetError(fmt.Errorf("%s failed to add JAS scan tasks: %s", logPrefix, generalError.Error()), auditParams.AllowPartialResults())
```

Comment:
This is a bug fix.
After adding the error to TargetError (or logging it if partial results are allowed) we left generalError as it was and it propagated up the stack and caught again later in the process. this caused the error to be logged twice.
After handling it, due to the new way we handle errors, there is no need to propagate the error by returning it after we handle it once